frontend/src/features/public-form/PublicFormService.ts

```diff
@@ -387,3 +452,41 @@ export const submitFormIssue = async (
     issueToPost,
   ).then(({ data }) => data)
 }
+
+/**
+ * Get presigned post data for attachments.
+ * @returns presigned post data for attachments.
+ */
+export const getAttachmentPresignedPostData = async ({
+  formFields,
+  formInputs,
+  formId,
+}: {
+  formFields: FormFieldDto[]
+  formInputs: FormFieldValues
+  formId: string
+}) => {
+  const attachmentsMap = getAttachmentsMap(formFields, formInputs)
+  const attachmentSizes: AttachmentSizeMapType[] = []
+  for (const id in attachmentsMap) {
+    // Check if id is a valid ObjectId. mongoose.isValidaObjectId cannot be used as it will throw a Reference Error.
+    const isValidObjectId = new RegExp(/^[0-9a-fA-F]{24}$/).test(id)
+    if (!isValidObjectId) throw new Error(`Invalid attachment id: ${id}`) // TODO: better error message?
```

Comment:
/sidepoint - can remove the TODO here, i think this error message is ok

Reply:
Done in [7c2d6e4](https://github.com/opengovsg/FormSG/pull/6767/commits/7c2d6e4088d7b6c616efd39333fcb7b7f3624877), thanks!